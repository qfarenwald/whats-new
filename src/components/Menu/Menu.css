.nav-topics {
  display: flex;
  justify-content: center;
  width: 100%;
}

h2 {
  margin: 20px;
}

h2:hover {
  cursor: pointer;
}
