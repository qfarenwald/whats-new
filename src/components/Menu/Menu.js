import React from 'react';
import './Menu.css'

const Menu = ( props ) => {
  const nav = props.topics.map((topic, i) => {
    return (
        <h2 id={topic} key={i} onClick={() => props.selectTopic(topic)}>{topic}</h2>
    )
  });

  return (
    <div className="nav-topics">
    {nav}
    </div>
  )
};

export default Menu;
