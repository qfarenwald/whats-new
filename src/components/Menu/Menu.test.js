import React from 'react';
import { shallow } from 'enzyme';
import Menu from './Menu'

describe('Menu', () => {
  it('should match snapshot with correct data', () => {
    const wrapper = shallow(
      <Menu
        topics={['local', 'entertainment', 'health', 'science', 'technology']}
        selectTopic={jest.fn()}
      />
    )
    expect(wrapper).toMatchSnapshot();
  })
})
