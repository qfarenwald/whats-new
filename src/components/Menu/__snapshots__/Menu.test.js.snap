// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu should match snapshot with correct data 1`] = `
<div
  className="nav-topics"
>
  <h2
    id="local"
    key="0"
    onClick={[Function]}
  >
    local
  </h2>
  <h2
    id="entertainment"
    key="1"
    onClick={[Function]}
  >
    entertainment
  </h2>
  <h2
    id="health"
    key="2"
    onClick={[Function]}
  >
    health
  </h2>
  <h2
    id="science"
    key="3"
    onClick={[Function]}
  >
    science
  </h2>
  <h2
    id="technology"
    key="4"
    onClick={[Function]}
  >
    technology
  </h2>
</div>
`;
